samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using SemanticKernel.Service.CopilotChat.Options;
+using SemanticKernel.Service.CopilotChat.Storage;
+
+namespace SemanticKernel.Service.CopilotChat.Skills.ChatSkills;
+
+/// <summary>
+/// This skill provides the functions to query the semantic chat memory.
+/// </summary>
+public class SemanticChatMemorySkill
+{
+    /// <summary>
+    /// Prompt settings.
+    /// </summary>
+    private readonly PromptsOptions _promptOptions;
+
+    /// <summary>
+    /// A repository to save and retrieve chat messages.
+    /// </summary>
+    private readonly ChatMessageRepository _chatMessageRepository;
+
+    /// <summary>
+    /// Create a new instance of SemanticChatMemorySkill.
+    /// </summary>
+    public SemanticChatMemorySkill(
+        IOptions<PromptsOptions> promptOptions,
+        ChatMessageRepository chatMessageRepository)
+    {
+        this._promptOptions = promptOptions.Value;
+        this._chatMessageRepository = chatMessageRepository;
+    }
+
+    /// <summary>
+    /// Query relevant memories based on the latest message.
+    /// </summary>
+    /// <param name="context">The SKContext</param>
+    /// <returns>A string containing the relevant memories.</returns>
+    [SKFunction("Query chat memories")]
+    [SKFunctionName("QueryChatMemories")]
+    [SKFunctionContextParameter(Name = "chatId", Description = "Chat ID to query history from")]
+    [SKFunctionContextParameter(Name = "tokenLimit", Description = "Maximum number of tokens")]
+    public async Task<string> QueryChatMemoriesAsync(SKContext context)
```

Comment:
rename to reflect query is based on last message. Maybe QueryLatestChatMemoriesAsync

Reply:
commenting to start a review.

Reply:
Good catch! I changed it to use user intent instead of the latest message to align with other skills/functions